keyboardClass.js

```diff
@@ -0,0 +1,241 @@
+
+            if (keyCode === 'ShiftLeft' || keyCode === 'ShiftRight') {
+                this.shiftPressed = false;
+                this.changeCase();
+                if (keyCode === 'ShiftLeft') this.btnList.ShiftRight.locked = false;
+                else this.btnList.ShiftLeft.locked = false;
+            } else if (keyCode === 'ControlLeft' || keyCode === 'ControlRight') {
+                if (this.ctrlPressed && this.altPressed) this.changeLanguage();
+                this.ctrlPressed = false;
+            } else if (keyCode === 'AltLeft' || keyCode === 'AltRight') {
+                if (this.ctrlPressed && this.altPressed) this.changeLanguage();
+                this.altPressed = false;
+            }
+        });
+        //
+        this.targetOfKeyboard.addEventListener('blur', () => {
+            this.posIn = this.targetOfKeyboard.selectionStart;
+        });
+        document.addEventListener('keyup', (event) => {
+            const keyCode = event.code ? event.code : event.keyCode;
+            if (!this.btnList[keyCode]) return;
+            if (keyCode === 'ShiftLeft' || keyCode === 'ShiftRight') {
+                this.shiftPressed = false;
+                this.changeCase();
+                if (keyCode === 'ShiftLeft') this.btnList.ShiftRight.locked = false;
+                else this.btnList.ShiftLeft.locked = false;
+            } else if (keyCode === 'ControlLeft' || keyCode === 'ControlRight') {
+                event.preventDefault();
+                this.ctrlPressed = false;
+            } else if (keyCode === 'AltLeft' || keyCode === 'AltRight') {
+                event.preventDefault();
+                this.altPressed = false;
+            }
+            this.btnList[keyCode].removeClass('active');
+        });
+        this.targetOfKeyboard.addEventListener('click', () => {
+            this.posIn = this.targetOfKeyboard.selectionStart;
+        });
+        this.totalNumberOfKeys = this.btnList.length;
+        this.rowNumber = 0;
+        this.numberOfKeysInRow = 0;
+        this.posIn = 0;
+        this.buttonListArray = [];
+        this.saveKeyboard();
+        this.changeLanguage.bind(this);
+    }
+
+    addBtn(button) {
+        this.totalNumberOfKeys += 1;
+        this.btnList[button.name] = button;
+        const rowsCounterConst = [14, 14, 14, 14, 14];
+        if (this.numberOfKeysInRow < rowsCounterConst[this.rowNumber]) {
+            this.keyboardRows[this.rowNumber].appendChild(button.domElement);
+        } else {
+            this.rowNumber += 1;
+            this.numberOfKeysInRow = 0;
+            this.keyboardRows[this.rowNumber].appendChild(button.domElement);
+        }
+        this.numberOfKeysInRow += 1;
+    }
+
+    triggerKeyboardEvent(el, keyState = 'keydown', keyCode) {
+        this.targetOfKeyboard.focus();
+        this.eventObj = new Event(keyState);
+        this.eventObj.keyCode = keyCode;
+        this.eventObj.which = keyCode;
+        el.dispatchEvent(this.eventObj);
```

Comment:
that's nice but why don't you assign `event.code` so it's the same as the real event